src/components/input/gamepadtokey.js

```diff
@@ -251,114 +251,150 @@ require(['apphost'], function (appHost) {
+                            case _GAMEPAD_B_BUTTON_INDEX:
+                                _ButtonPressedState.setgamepadB(true);
+                                break;
+                            default:
+                                // No-op
+                                break;
+                        }
+                    } else {
+                        switch (j) {
+                            case _GAMEPAD_DPAD_UP_BUTTON_INDEX:
+                                if (_ButtonPressedState.getdPadUp()) {
+                                    _ButtonPressedState.setdPadUp(false);
+                                }
+                                break;
+                            case _GAMEPAD_DPAD_DOWN_BUTTON_INDEX:
+                                if (_ButtonPressedState.getdPadDown()) {
+                                    _ButtonPressedState.setdPadDown(false);
+                                }
+                                break;
+                            case _GAMEPAD_DPAD_LEFT_BUTTON_INDEX:
+                                if (_ButtonPressedState.getdPadLeft()) {
+                                    _ButtonPressedState.setdPadLeft(false);
+                                }
+                                break;
+                            case _GAMEPAD_DPAD_RIGHT_BUTTON_INDEX:
+                                if (_ButtonPressedState.getdPadRight()) {
+                                    _ButtonPressedState.setdPadRight(false);
+                                }
+                                break;
+                            case _GAMEPAD_A_BUTTON_INDEX:
+                                if (_ButtonPressedState.getgamepadA()) {
+                                    _ButtonPressedState.setgamepadA(false);
+                                }
+                                break;
+                            case _GAMEPAD_B_BUTTON_INDEX:
+                                if (_ButtonPressedState.getgamepadB()) {
+                                    _ButtonPressedState.setgamepadB(false);
+                                }
+                                break;
+                            default:
+                                // No-op
+                                break;
                         }
                     }
                 }
             }
         }
         // Schedule the next one
-        requestAnimationFrame(runInputLoop);
+        inputLoopTimer = requestAnimationFrame(runInputLoop);
     }
 
-    runInputLoop();
+    function startInputLoop() {
+        if (!inputLoopTimer) {
+            runInputLoop();
+        }
+    }
+
+    function stopInputLoop() {
+        cancelAnimationFrame(inputLoopTimer);
+        inputLoopTimer = undefined;
+    }
+
+    function isGamepadConnected() {
+        var gamepads = navigator.getGamepads();
+        for (var i = 0, len = gamepads.length; i < len; i++) {
+            var gamepad = gamepads[i];
+            if (gamepad) {
```

Comment:
```suggestion
            if (gamepad && gamepad.connected) {
```

Reply:
So I suppose that you checked that the field exists as well in Chrome, right?

Reply:
Yes. At least, Chrome 79.0.3945.79 has it.
Also, I checked Opera 66.0.3515.115 - OK
Edge does not receive gamepads.